### ISSUE TITLE
expectTranslation test helper not working with ember-cli 2.9.0

### ISSUE THREAD

**Issue by thagul** [jamesarosen/ember-i18n#413]
ember-cli v2.9.0 includes a version bump of QUnit to 2.0.0 which doesn't support global `ok`. Currently `expectTranslation` test helper uses global `ok` and it throws errors when upgrading to ember-cli 2.9.0 and then QUnit to 2.0.0.

Current version is 

```
if (typeof QUnit !== 'undefined' && typeof ok === 'function') {
  return function(element, key, text) {
    ok(find(`${element}:contains(${text})`).length, `Found translation key ${key} in ${element}`);
  };
}
```

As per the [QUnit upgrade guide](https://qunitjs.com/upgrade-guide-2.x/) I think that it should be changed to 

```
if (typeof QUnit !== 'undefined' && typeof QUnit.assert.ok === 'function') {
  return function(element, key, text) {
    QUnit.assert.ok(find(`${element}:contains(${text})`).length, `Found translation key ${key} in ${element}`);
  };
}
```

**Comment by jamesarosen** [comment]
You'll need to support `window.ok` for older QUnit users. Try
```js
(function() {
  if (typeof QUnit === 'undefined') return;
  
  const ok = Ember.get(window, 'ok') || Ember.get(QUnit, 'assert.ok');
  if (ok !== 'function') return;

  return function(element, key, text) {
    ok(...);
  };
}());
```

**Comment by thagul** [comment]
I'm sorry, maybe I wasn't detailed enough describing the issue, but your suggestion doesn't work.
`window.ok` is still defined also in QUnit 2 as a function that throws an error
```
throw new Error(
  "ok is removed in QUnit 2.0.\n" +
  "Details in our upgrade guide at https://qunitjs.com/upgrade-guide-2.x/"
);
```

So `const ok = Ember.get(window, 'ok') || Ember.get(QUnit, 'assert.ok');` will always return `window.ok`.
I've tried reverting elements in the `or`

`const ok = Ember.get(QUnit, 'assert.ok') || Ember.get(window, 'ok');`

but this way when calling `ok(...)` to verify the assertion, qunit throws another error (this.pushResult is undefined).

I've tried with following and it seems to work but, unfortunately, I haven't enough knowledge to confirm that this could be the right solution
```
const assertTranslation = (function() {

  if (typeof QUnit !== 'undefined' && (typeof Ember.get(window, 'ok') === 'function' || typeof Ember.get(QUnit, 'assert.ok') === 'function')) {
    let assert = Ember.get(QUnit, 'assert') || {};
    if (typeof assert.ok !== 'function') {
      assert.ok = Ember.get(window.ok);
    }
    return function(element, key, text) {
      assert.ok(find(`${element}:contains(${text})`).length, `Found translation key ${key} in ${element}`);
    };
  } else if (typeof expect === 'function') {
    return function(element, key, text) {
      const found = !!(find(`${element}:contains(${text})`).length);
      expect(found).to.equal(true);
    };
  } else {
    return function() {
      throw new Error("ember-i18n could not find a compatible test framework");
    };
  }
}());
```

**Comment by mlenderink** [comment]
Same issue here while on Ember 2.9 as a temporary fix i did this

```javascript
window.ok = function() {
    QUnit.assert.push(...arguments);
}
```

**Comment by barryofguilder** [comment]
@mlenderink, where did you put that temporary fix at? We're wanting to upgrade our project to Ember CLI 2.10 and I need to get around this issue.

**Comment by mlenderink** [comment]
@barryofguilder I put that code in our custom-test-helpers file. It should work if you put it in the `tests/helpers/start-app.js` file

**Comment by barryofguilder** [comment]
@mlenderink Thanks, that worked!

**Comment by xtagon** [comment]
This didn't work for me. When I try that workaround I get `'QUnit' is not defined.`. Any ideas?

**Comment by xtagon** [comment]
I solved `'QUnit' is not defined.` by explicitly importing QUnit into `tests/helpers/start-app.js` before the workaround, but I still get `'expectTranslation' is not defined`.

Ember version 2.10.1

**Comment by barryofguilder** [comment]
I'm not sure what happened, but after doing an `npm install` now, I get the error again that says "ember-i18n could not find a compatible test framework". For some reason, the fix being in the `tests/helpers/start-app.js` file is no longer working.

I fixed this by moving it to the `tests/test-helper.js` file. Not sure what made this break, but this seems to work... for now.

So to recap, I added this to my `tests/test-helper.js` file:

```js
import QUnit from 'qunit';

window.ok = function() {
  QUnit.assert.push(...arguments);
};
```
